vm/vm.go

```diff
@@ -37,6 +38,8 @@ type VM struct {
 	containerID      string                    `yaml:"container-id"`
 	NetOpts          vmTypes.NetworkingOptions `yaml:"networking"`
 	Shares           []string                  `yaml:"shares"`
+	ExposedPorts     []string                  `yaml:"ports"`
+	PortOpts         vmTypes.PortOptions
```

Comment:
let's keep this `VM` structure more generic in order to to allow multiple container/vm engines in the future and not being tied to docker-based networking options.